src/lib/change-handlers.ts

```diff
@@ -0,0 +1,160 @@
+/*!
+ * The contents of this file are subject to the Common Public Attribution License Version 1.0 (the “License”); you may not use this file except in compliance with the License. You may obtain a copy of the License at https://mpapp-public.gitlab.io/manuscripts-frontend/LICENSE. The License is based on the Mozilla Public License Version 1.1 but Sections 14 and 15 have been added to cover use of software over a computer network and provide for limited attribution for the Original Developer. In addition, Exhibit A has been modified to be consistent with Exhibit B.
+ *
+ * Software distributed under the License is distributed on an “AS IS” basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the specific language governing rights and limitations under the License.
+ *
+ * The Original Code is manuscripts-frontend.
+ *
+ * The Original Developer is the Initial Developer. The Initial Developer of the Original Code is Atypon Systems LLC.
+ *
+ * All portions of the code written by Atypon Systems LLC are Copyright (c) 2024 Atypon Systems LLC. All Rights Reserved.
+ */
+import {
+  NodeAttrChange,
+  NodeChange,
+  TextChange,
+} from '@manuscripts/track-changes-plugin'
+import { schema } from '@manuscripts/transform'
+
+import { NodeTextContentRetriever } from './node-content-retriever'
+import { changeOperationAlias } from './tracking'
+import { getParentNode } from './utils'
+
+interface SnippetData {
+  operation: string
+  nodeName: string
+  content: string
+  isEquation?: boolean
+}
+
+export const handleTextChange = (
+  suggestion: TextChange,
+  view: any,
+  dataTracked: any
+): SnippetData | null => {
+  const parentNodeType = getParentNode(view.state, suggestion.from)?.type
+  const parentNodeName = parentNodeType?.spec.name || parentNodeType?.name
+  const nodeName =
+    parentNodeType === schema.nodes.paragraph
+      ? 'text'
+      : parentNodeName + ' text'
+
+  return {
+    operation: changeOperationAlias(dataTracked.operation),
+    nodeName: nodeName || suggestion.nodeType.name,
+    content: suggestion.text,
+  }
+}
+
+export const handleNodeChange = (
+  suggestion: NodeChange | NodeAttrChange,
+  view: any,
+  doc: any,
+  dataTracked: any
+): SnippetData | null => {
+  const nodeContentRetriever = new NodeTextContentRetriever(view.state)
+
+  switch (suggestion.node.type) {
+    case schema.nodes.inline_footnote: {
+      return {
+        operation: changeOperationAlias(dataTracked.operation),
```

Comment:
Destructure `nodeName` and `operation` for cleaner code and to avoid repetitive access. Add these lines just before the switch statement:
```
const { node } = suggestion
const operation = changeOperationAlias(dataTracked.operation)
const nodeName = node.type.spec.name || node.type.name
```

Reply:
done

Reply:
Just pass `operation` and `nodeName` directly; there's no need for `operation: operation` and `nodeName: nodeName`. Also, I don't see you using `node` instead of `suggestion.node`.

Reply:
done